core/src/main/java/org/kiwix/kiwixmobile/core/utils/SharedPreferenceUtil.kt

```diff
@@ -0,0 +1,182 @@
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/**
+ * Manager for the Default Shared Preferences of the application.
+ */
+@Singleton
+class SharedPreferenceUtil @Inject constructor(context: Context?) {
+  private val sharedPreferences: SharedPreferences =
+    PreferenceManager.getDefaultSharedPreferences(context)
+  val prefStorages = PublishProcessor.create<String>()
+  private val textZooms = PublishProcessor.create<Int>()
+  private val nightModes = PublishProcessor.create<NightModeConfig.Mode>()
+
+  val prefWifiOnly: Boolean
+    get() = sharedPreferences.getBoolean(PREF_WIFI_ONLY, true)
+
+  val prefHideToolbar: Boolean
+    get() = sharedPreferences.getBoolean(PREF_HIDE_TOOLBAR, true)
+
+  val prefIsFirstRun: Boolean
+    get() = sharedPreferences.getBoolean(PREF_IS_FIRST_RUN, true)
+
+  val prefFullScreen: Boolean
+    get() = sharedPreferences.getBoolean(PREF_FULLSCREEN, false)
+
+  val prefBackToTop: Boolean
+    get() = sharedPreferences.getBoolean(PREF_BACK_TO_TOP, false)
+
+  val prefNewTabBackground: Boolean
+    get() = sharedPreferences.getBoolean(PREF_NEW_TAB_BACKGROUND, false)
+
+  val prefExternalLinkPopup: Boolean
+    get() = sharedPreferences.getBoolean(PREF_EXTERNAL_LINK_POPUP, true)
+
+  val prefLanguage: String
+    get() = sharedPreferences.getString(PREF_LANG, Locale.ROOT.toString())
+
+  val prefStorage: String
+    get() {
+      val storage = sharedPreferences.getString(PREF_STORAGE, null)
+      if (storage == null) {
+        val defaultStorage = defaultStorage()
+        putPrefStorage(defaultStorage)
+        return defaultStorage().also(this@SharedPreferenceUtil::putPrefStorage)
+      } else if (!File(storage).exists()) {
+        return defaultStorage()
+      }
+      return storage
+    }
+
+  private fun defaultStorage(): String =
+    /*val externalFilesDir = ContextCompat.getExternalFilesDirs(instance, null)[0]
+    return if (externalFilesDir != null)
+      externalFilesDir.path
+    else
+      instance.filesDir.path */ // workaround for emulators
+    ((getExternalFilesDirs(instance, null)[0]))?.let { it.path ?: instance.filesDir.path }!!
+
+  fun getPrefStorageTitle(defaultTitle: String): String =
+    sharedPreferences.getString(PREF_STORAGE_TITLE, defaultTitle)
+
+  fun putPrefLanguage(language: String?) =
+    sharedPreferences.edit().putString(PREF_LANG, language).apply()
+
+  fun putPrefIsFirstRun(isFirstRun: Boolean) =
+    sharedPreferences.edit().putBoolean(PREF_IS_FIRST_RUN, isFirstRun).apply()
+
+  fun putPrefWifiOnly(wifiOnly: Boolean) =
+    sharedPreferences.edit().putBoolean(PREF_WIFI_ONLY, wifiOnly).apply()
+
+  fun putPrefStorageTitle(storageTitle: String?) =
+    sharedPreferences.edit().putString(PREF_STORAGE_TITLE, storageTitle).apply()
+
+  fun putPrefStorage(storage: String) {
+    sharedPreferences.edit().putString(PREF_STORAGE, storage).apply(); prefStorages.onNext(storage)
+  }
+
+  fun getPrefStorages(): Flowable<String> = prefStorages.startWith(prefStorage)
```

Comment:
![image](https://user-images.githubusercontent.com/3358282/87522607-bd751500-c67d-11ea-9661-52fbcb379b6f.png)


This method is now unused because of your changes which was unintentional. It is imperative to check the icons in the gutter to the right and remove as many errors as possible from a file